packages/core/src/editor/transformPasted.ts

```diff
@@ -49,31 +51,45 @@ export function wrapTableRows(f: Fragment, schema: Schema) {
 /**
  * fix for https://github.com/ProseMirror/prosemirror/issues/1430#issuecomment-1822570821
  *
- * Without this fix, pasting two paragraphs would cause the second one to be indented in the other
- * this fix wraps every element in the slice in it's own blockContainer, to prevent Prosemirror from nesting the
- * elements on paste.
+ * This fix wraps pasted ProseMirror nodes in their own `blockContainer` nodes
+ * in some cases. This is to ensure that ProseMirror inserts them as separate
+ * blocks, which it sometimes doesn't do because it doesn't have enough context
+ * about the hierarchy of the pasted nodes. This can be seen when pasting e.g.
+ * an image or two consecutive paragraphs, where PM tries to nest the pasted
+ * block(s) when it shouldn't. However, this fix is not applied in a few cases.
+ *
+ * The first case is when we paste a single node with inline content, e.g. a
+ * paragraph or heading. This is because we want to insert the content in-line
+ * for better UX instead of a separate block below.
  *
- * The exception is when we encounter blockGroups with listitems, because those actually should be nested
+ * The second case is when we paste a single node with table content, i.e. a
+ * table, inside an existing table. This is because the fix would tell instead
```

Comment:
typo? `This is because the fix would tell instead want to add its content to the existing table, again for better UX.`